cypress/integration/providers_spec.js

```diff
@@ -0,0 +1,91 @@
+
+      cy.get('table tbody tr').its('length').should('be.gt', 1);
+    });
+
+    it('providers page displays a button to add a new provider', () => {
+      cy.visit(`${host}/#providers`);
+
+      cy.get('.heading--large').should('have.text', 'Provider Overview');
+      cy.get('a').contains('Add a Provider').should('exist').as('addProvider');
+      cy.get('@addProvider').should('have.attr', 'href', '#/providers/add');
+      cy.get('@addProvider').click();
+
+      cy.get('.heading--xlarge').should('have.text', 'Providers');
+      cy.get('.heading--large').should('have.text', 'Create a provider');
+
+      // fill the form and submit
+      // we need to use fake provider under test/fake-api-fixtures/providers, so the
+      // verification would work
+      cy.get('form div ul').as('providerinput');
+      cy.get('@providerinput').contains('Provider Name').siblings('input').type('s3_provider');
+      cy.get('@providerinput').contains('Concurrent Connnection Limit').siblings('input').clear().type(5);
+      cy.get('@providerinput').contains('Protocol').siblings().children('select').select('s3').should('have.value', 's3');
+      cy.get('@providerinput').contains('Host').siblings('input').type('cumulus-test-sandbox-internal');
+
+      cy.get('form div input[value=Submit]').click();
+
+      // displays the new provider
+      cy.get('.heading--xlarge').should('have.text', 'Providers');
+      cy.get('.heading--large').should('have.text', 's3_provider');
+      cy.url().should('include', '#/providers/provider/s3_provider');
+    });
+
+    it('provider page has buttons to edit or delete the provider', () => {
+      cy.visit(`${host}/#/providers/provider/s3_provider`);
+      cy.get('.heading--large').should('have.text', 's3_provider');
+      cy.get('a').contains('Edit').should('exist').as('editprovider');
+      cy.get('@editprovider').should('have.attr', 'href').and('include', '#/providers/edit/s3_provider');
+      cy.get('@editprovider').click();
+
+      cy.get('.heading--large').should('have.text', 'Edit s3_provider');
+
+      cy.get('form div ul').as('providerinput');
+      cy.get('@providerinput').contains('Concurrent Connnection Limit').siblings('input').clear().type(12);
+      cy.get('@providerinput').contains('Host').siblings('input').clear().type('cumulus-test-sandbox-internal-new');
+
+      cy.get('form div input[value=Submit]').click();
+
+      // displays the updated provider
+      cy.get('.heading--xlarge').should('have.text', 'Providers');
+      cy.get('.heading--large').should('have.text', 's3_provider');
+
+      // delete provider
+      cy.get('.dropdown__options__btn').click();
+      cy.get('span').contains('Delete').click();
+      cy.get('button').contains('Confirm').click();
+
+      // goes back to providers list
+      cy.url().should('include', 'providers');
+      cy.get('.heading--xlarge').should('have.text', 'Providers');
+      cy.get('table tbody tr').its('length').should('be.gt', 1);
```

Comment:
Does this actually prove that the provider was deleted?

Reply:
How do we know for sure that the providers list should still be greater than 1 at this point?

Reply:
I see, it is because the fixture for providers has 2 items? In which case, we could test for equality to 2 exactly. But I'm wondering whether it's even important to test the list length as long as we can assert that the row for the specific provider we deleted does not exist